# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/ipc/PWebGPU.ipdl
# Commit: 843c7d19787b
# Full Hash: 843c7d19787b5beb9caa2fbf71192b174738bb92
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 1
# Description:
#   Bug 1838693 Part 1: Stub in 'lose the device' and trigger it on device lost errors. r=webgpu-reviewers,nical
#   
#   This creates a WebGPUParent::LoseDevice entry point, which informs the
#   WebGPUChild that the device has been lost. The child side is largely a
#   stub, as a later part will rationalize the handling of the device.lost
# ==============================================================================

diff -r e0dd0b10e8fd -r 843c7d19787b dom/webgpu/ipc/PWebGPU.ipdl
--- a/dom/webgpu/ipc/PWebGPU.ipdl	Mon Oct 16 15:29:21 2023 +0000
+++ b/dom/webgpu/ipc/PWebGPU.ipdl	Mon Oct 16 15:31:59 2023 +0000
@@ -89,6 +89,7 @@
 child:
   async UncapturedError(RawId? aDeviceId, nsCString message);
   async DropAction(ByteBuf buf);
+  async DeviceLost(RawId aDeviceId, uint8_t? reason, nsCString message);
   async __delete__();
 };
 